ports/esp32/machine_wdt.c

```diff
@@ -42,19 +43,40 @@ typedef struct _machine_wdt_obj_t {
 STATIC machine_wdt_obj_t wdt_default = {{&machine_wdt_type}};
 
 STATIC mp_obj_t machine_wdt_make_new(const mp_obj_type_t *type_in, size_t n_args, size_t n_kw, const mp_obj_t *args) {
-    mp_arg_check_num(n_args, n_kw, 0, 1, false);
+    mp_arg_check_num(n_args, n_kw, 0, 2, true);
+    
+    enum {
+        ARG_timeout, ARG_panic
+    };
 
-    mp_int_t id = 0;
-    if (n_args > 0) {
-        id = mp_obj_get_int(args[0]);
-    }
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_timeout, MP_ARG_INT, {.u_int = 0} },
+        { MP_QSTR_panic, MP_ARG_BOOL, {.u_bool = true} }
+    };
+
+    mp_map_t kw_args;
+    mp_arg_val_t mp_args[MP_ARRAY_SIZE(allowed_args)];
+
+    mp_map_init_fixed_table(&kw_args, n_kw, args + n_args);
+    mp_arg_parse_all(n_args, args, &kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, mp_args);
+
+    if (mp_args[ARG_timeout].u_int == 0) {
```

Comment:
I think it might be easier and less code if you simply put sensible defaults for the keyword arguments and then unconditionally called `esp_task_wdt_init()` with whatever values you get from the keywords.

Reply:
I took as an example the file machine_uart.c to understand how to take the keyword arguments. What specifically does the `mp_map_init_fixed_table()` method do?

Reply:
I'd tried to pin my comment on line 63; I'm not sure why it came out above.

What I was meaning is that if you set the default value for `timeout` to something other than 0 (as discussed below) then you can skip the whole of the True side of the `if (mp_args[ARG_timeout].u_int == 0)` statement and go strait to the code that is currently in the `else` side.